pkg/domain/infra/abi/images.go

```diff
@@ -725,63 +721,66 @@ func (ir *ImageEngine) Sign(ctx context.Context, names []string, options entitie
 		if dockerReference == nil {
 			return nil, errors.Errorf("cannot determine canonical Docker reference for destination %s", transports.ImageName(rawSource.Reference()))
 		}
-
-		// create the signstore file
-		rtc, err := ir.Libpod.GetConfig()
-		if err != nil {
-			return nil, err
-		}
-		newImage, err := ir.Libpod.ImageRuntime().New(ctx, signimage, rtc.Engine.SignaturePolicyPath, "", os.Stderr, &dockerRegistryOptions, image.SigningOptions{SignBy: options.SignBy}, nil, util.PullImageMissing)
-		if err != nil {
-			return nil, errors.Wrapf(err, "error pulling image %s", signimage)
+		var sigStoreDir string
+		if options.Directory != "" {
+			sigStoreDir = options.Directory
 		}
 		if sigStoreDir == "" {
 			if rootless.IsRootless() {
 				sigStoreDir = filepath.Join(filepath.Dir(ir.Libpod.StorageConfig().GraphRoot), "sigstore")
 			} else {
+				var sigStoreURI string
 				registryInfo := trust.HaveMatchRegistry(rawSource.Reference().DockerReference().String(), registryConfigs)
 				if registryInfo != nil {
-					if sigStoreDir = registryInfo.SigStoreStaging; sigStoreDir == "" {
-						sigStoreDir = registryInfo.SigStore
+					if sigStoreURI = registryInfo.SigStoreStaging; sigStoreURI == "" {
+						sigStoreURI = registryInfo.SigStore
+					}
+				}
 
+				if sigStoreURI == "" {
+					if registryConfigs.DefaultDocker == nil {
+						logrus.Debugf(" No signature storage configuration found for %s", rawSource.Reference().DockerReference().String())
+						continue
+					}
+					if sigStoreURI = registryConfigs.DefaultDocker.SigStoreStaging; sigStoreURI == "" {
```

Comment:
I was expecting `HaveMatchRegistry` to handle the `DefaultDocker` setting — apparently it doesn’t, but I maybe it should be modified that way, also for `podman trust show` to correctly (?) report the right signature store path in the default case.

Either way, it would be simpler to have a single `if registryInfo == nil { registryInfo = registryConfigs.DefaultDocker }` check and a single `SigStoreStaging`/`SigStore` decision.